plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -408,22 +406,51 @@ public function get_views() {
 	/**
 	 * Count orders by status.
 	 *
-	 * @param string $status The order status we are interested in.
+	 * @param string|string[] $status The order status we are interested in.
 	 *
 	 * @return int
 	 */
-	private function count_orders_by_status( string $status ): int {
+	private function count_orders_by_status( $status ): int {
 		$orders = wc_get_orders(
```

Comment:
We should probably replace this with direct count query, it definitely won't work on large sites.

Reply:
You're right. I've changed this to use `wc_orders_count()` instead, which uses the underlying datastore for counting and is also cached. I have no problem with writing a count query but thought it'd be best to use our own APIs if those worked here.

The only drawback I see is that it doesn't count multiple statuses at once, but that shouldn't be too bad (as there aren't that many statuses). Also, we need the individual count later so caching covers that.

If you still think I should go with a direct query, I'd be more than happy to do that.